sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -792,7 +780,7 @@ case class GpuHashAggregateExec(
         // desired result compared to Spark-CPU.
         // For more details go to https://github.com/NVIDIA/spark-rapids/issues/1737
         if (cvs.isEmpty) {
-          withResource(Scalar.fromLong(0L)) { ZERO =>
+          withResource(GpuScalar.from(0L, LongType)) { ZERO =>
```

Comment:
OK, I reverted all of them.
Currently both the `GpuScalar.from` and `cudf.Scalar.fromXXX` are widely used. 
I did this because it is suggested as "_I really would like to see a separate PR that has us stop using any of the Scalar factory APIs outside of GpuScalar itself"_ in this comment https://github.com/NVIDIA/spark-rapids/pull/2313#pullrequestreview-650266129

Reply:
`GpuScalar.from` is needed when we don't know at compile-time what type we need.  If we know the type at compile-time then `Scalar.from*` is better since the value argument is type-checked at compile time and avoids the runtime type dispatch.  Therefore I don't think it makes sense to say "all one or the other" because they have different use-cases.

Reply:
Make sense